lib/include/tick/base/report.h

```diff
@@ -0,0 +1,51 @@
+//
+//  report.h
+//  TICK
+//
+//  Created by Philip Deegan
+//  Copyright (c) 2015 bacry. All rights reserved.
+//
+//  Multipurpose Wrapper return type
+
+#ifndef LIB_INCLUDE_TICK_BASE_REPORT_H_
+#define LIB_INCLUDE_TICK_BASE_REPORT_H_
+
+#include <sstream>
+
+template <class A = bool, class B = std::string>
+class Report {};
+
+class BoolStrReport : public Report<bool, std::string> {
+  template <class T1>
+  friend std::ostream& operator<<(std::ostream&, const BoolStrReport&);
+
+ private:
+  bool success = false;
+  std::string str;
+
+ public:
+  BoolStrReport(bool b, const std::string& s) : success(b), str(s) {}
+  BoolStrReport(BoolStrReport& b, const std::string& s)
+      : success(b.success), str(s) {}
+  explicit BoolStrReport(const std::pair<bool, std::string>& p)
+      : success(p.first), str(p.second) {}
+  explicit BoolStrReport(const std::pair<BoolStrReport, std::string>& p)
```

Comment:
With this constructor you completely ignore the `str` attribute of the BoolStrReport, is this wanted?

Reply:
Oh, I think I understand, you will report the first one which fails? Perhaps this could be told explicitly somewhere :).